frame/staking/src/lib.rs

```diff
@@ -1765,9 +1765,7 @@ impl <T: Trait> OnOffenceHandler<T::AccountId, pallet_session::historical::Ident
 		};
 
 		<Self as Store>::EarliestUnappliedSlash::mutate(|earliest| {
-			if earliest.is_none() {
-				*earliest = Some(era_now)
-			}
+			*earliest = Some(earliest.unwrap_or(era_now).min(era_now));
```

Comment:
The previous logic seems correct - the era hasn't moved backwards since the last time `EarliestUnappliedSlash` was set and we have `apply_unapplied_slashes` for moving `EarliestUnappliedSlash` forward.

Reply:
You could rewrite this block as

```rust
*earliest = Some(match *earliest {
    None => era_now,
    Some(prev) => prev,
})
```

which is the same as the previous logic.